partest/src/main/scala/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -74,19 +73,9 @@ class MainGenericRunner {
         useClosureCompiler = optMode == FullOpt)
 
     val libJSEnv = {
-      /* Historically, we used Rhino in NoOpt and NodeJS in FastOpt and FullOpt.
-       * This is not necessary anymore: Rhino can run in FastOpt.
-       * We keep this for now, mainly to not change too many things at once.
-       */
-      if (optMode == NoOpt) {
-        val env = new RhinoJSEnv(semantics).withSourceMap(false)
-        val unit = linker.linkUnit(ir, env.symbolRequirements, logger)
-        env.loadLinkingUnit(unit)
-      } else {
-        val output = WritableMemVirtualJSFile("partest-fastOpt.js")
-        linker.link(ir, output, logger)
-        new NodeJSEnv().loadLibs(ResolvedJSDependency.minimal(output) :: Nil)
-      }
+      val output = WritableMemVirtualJSFile("partest-fastopt.js")
+      linker.link(ir, output, logger)
+      new NodeJSEnv().loadLibs(ResolvedJSDependency.minimal(output) :: Nil)
     }
 
     libJSEnv.jsRunner(jsRunner).run(logger, jsConsole)
```

Comment:
Replace this with:

```scala
val sjsCode = {
  val output = WritableMemVirtualJSFile("partest.js")
  linker.link(ir, output, logger)
  ResolvedJSDependency.minimal(output) :: Nil
}

new NodeJSEnv().jsRunner(sjsCode, jsRunner).run(logger, jsConsole)
```

Seems simpler to understand to not split loading of lib and runner code... Now that I realize that it is actually not less code, your call I guess.